### PR TITLE
Statements reference: minor edits

### PR DESCRIPTION
**Pull request by pkulikov**
---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/statements/checked-and-unchecked.md](https://github.com/dotnet/docs/blob/01141bdf8b4bef490cf90725069ce2a73788e9f3/docs/csharp/language-reference/statements/checked-and-unchecked.md) | [checked and unchecked statements (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/checked-and-unchecked?branch=pr-en-us-35183) |
| [docs/csharp/language-reference/statements/fixed.md](https://github.com/dotnet/docs/blob/01141bdf8b4bef490cf90725069ce2a73788e9f3/docs/csharp/language-reference/statements/fixed.md) | [fixed statement - safely access memory underlying a variable](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/fixed?branch=pr-en-us-35183) |
| [docs/csharp/language-reference/statements/iteration-statements.md](https://github.com/dotnet/docs/blob/01141bdf8b4bef490cf90725069ce2a73788e9f3/docs/csharp/language-reference/statements/iteration-statements.md) | [Iteration statements - `for`, `foreach`, `do`, and `while`](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/iteration-statements?branch=pr-en-us-35183) |
| [docs/csharp/language-reference/statements/jump-statements.md](https://github.com/dotnet/docs/blob/01141bdf8b4bef490cf90725069ce2a73788e9f3/docs/csharp/language-reference/statements/jump-statements.md) | [Jump statements - `break`, `continue`, `return`, and `goto`](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/jump-statements?branch=pr-en-us-35183) |
| [docs/csharp/language-reference/statements/lock.md](https://github.com/dotnet/docs/blob/01141bdf8b4bef490cf90725069ce2a73788e9f3/docs/csharp/language-reference/statements/lock.md) | [lock statement - ensure exclusive access to a shared resource](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock?branch=pr-en-us-35183) |
| [docs/csharp/language-reference/statements/selection-statements.md](https://github.com/dotnet/docs/blob/01141bdf8b4bef490cf90725069ce2a73788e9f3/docs/csharp/language-reference/statements/selection-statements.md) | [Selection statements - `if`, `else`, and `switch`](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/selection-statements?branch=pr-en-us-35183) |
| [docs/csharp/language-reference/statements/yield.md](https://github.com/dotnet/docs/blob/01141bdf8b4bef490cf90725069ce2a73788e9f3/docs/csharp/language-reference/statements/yield.md) | [yield statement - provide the next element](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/yield?branch=pr-en-us-35183) |

### CODE REVIEW

File: docs/csharp/language-reference/statements/fixed.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ helpviewer_keywords:
   - "fixed statement [C#]"
   - "fixed keyword [C#]"
 ---
-# fixed statement - safely access memory underlying a variable
+# fixed statement - pin a variable for pointer operations
```
Comment (pkulikov) [inline docs/csharp/language-reference/statements/fixed.md:12]: Original text may give an impression that by default, without `fixed`, you cannot safely access variable. In the new version, I also would like to point that `fixed` is for scenarios with pointers, so most probably, a rare use.

File: docs/csharp/language-reference/statements/jump-statements.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ helpviewer_keywords:
...
+The jump statements unconditionally transfer control. The [`break` statement](#the-break-statement) terminates the closest enclosing [iteration statement](iteration-statements.md) or [`switch` statement](selection-statements.md#the-switch-statement). The [`continue` statement](#the-continue-statement) starts a new iteration of the closest enclosing [iteration statement](iteration-statements.md). The [`return` statement](#the-return-statement) terminates execution of the function in which it appears and returns control to the caller. The [`goto` statement](#the-goto-statement) transfers control to a statement that is marked by a label.
```
Comment (pkulikov) [inline docs/csharp/language-reference/statements/jump-statements.md:22]: Removed the sentence about `ref` as it looks out of context in this paragraph that gives the general overview of the statements. I assume that ref returns are not often met.

File: docs/csharp/language-reference/statements/snippets/checked-and-unchecked/Program.cs [hunk 3]
```diff
@@ -9,14 +9,14 @@ void MainExample()
 
     unchecked
     {
-        Console.WriteLine(a + 1);  // output: 0
+        Console.WriteLine(a + 3);  // output: 2
```
Comment (pkulikov) [inline docs/csharp/language-reference/statements/snippets/checked-and-unchecked/Program.cs:12]: Want to show that overflow doesn't falls back to the minimum possible value, but really wraps

**Review by pkulikov — COMMENTED** [review]
Fixed typos, grammar and style fixes; also tried to simplify some sentences.

File: docs/csharp/language-reference/statements/fixed.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/language-reference/statements/fixed.md:12]: I like it. That's a great edit.

**Review by BillWagner — APPROVED** [review]
Thanks again @pkulikov 

These all look great. I'll :shipit: now.

**Pull request merged**